build/cluster.tf

```diff
@@ -67,15 +73,15 @@ locals {
 # before creating the cluster to avoid 
 # unfinished configurations
 resource "null_resource" "check-password-length" {
-  count = "${length(var.password) >= 16 || length(var.password) == 0 ? 0 : 1}"
-  "Password must be more than 16 chars in length" = true
+  count = "${length(var.password) >= 16 || length(var.password) == 0 ? 0 : "P"}" #Error: Password must be more than or equal to 16 chars in length
 }
 
 resource "google_container_cluster" "primary" {
-  name     = "${lookup(var.cluster, "name")}"
-  location     = "${lookup(var.cluster, "zone")}"
-  project  = "${lookup(var.cluster, "project")}"
-  provider = "google-beta"
+  name               = "${var.cluster["name"]}"
+  location           = "${var.cluster["zone"]}"
+  project            = "${var.cluster["project"]}"
+  provider           = "google-beta"
+  enable_legacy_abac = "${var.cluster["legacyAbac"]}"
 
   # Setting an empty username and password explicitly disables basic auth
```

Comment:
Please remove the master_auth block. Now that Mark switched GKE to 1.12 the defaults in GKE disable both basic auth and client certs.

Reply:
Thanks for the information, will update soon